launchdarkly/resource_launchdarkly_view.go

```diff
@@ -0,0 +1,245 @@
+	return &schema.Resource{
+		CreateContext: resourceViewCreate,
+		ReadContext:   resourceViewRead,
+		UpdateContext: resourceViewUpdate,
+		DeleteContext: resourceViewDelete,
+		Exists:        resourceViewExists,
+
+		Importer: &schema.ResourceImporter{
+			StateContext: resourceViewImport,
+		},
+
+		Description: `Provides a LaunchDarkly view resource.
+
+This resource allows you to create and manage views within your LaunchDarkly project.`,
+
+		Schema: map[string]*schema.Schema{
+			PROJECT_KEY: {
+				Type:             schema.TypeString,
+				Required:         true,
+				Description:      addForceNewDescription("The project key.", true),
+				ForceNew:         true,
+				ValidateDiagFunc: validateKey(),
+			},
+			KEY: {
+				Type:             schema.TypeString,
+				Required:         true,
+				Description:      addForceNewDescription("The view's unique key.", true),
+				ForceNew:         true,
+				ValidateDiagFunc: validateKey(),
+			},
+			NAME: {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "The view's name.",
+			},
+			DESCRIPTION: {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "The view's description.",
+			},
+			GENERATE_SDK_KEYS: {
+				Type:        schema.TypeBool,
+				Optional:    true,
+				Default:     false,
+				Description: "Whether to generate SDK keys for this view.",
+			},
+			MAINTAINER_ID: {
+				Type:          schema.TypeString,
+				Optional:      true,
+				Description:   "The member ID of the maintainer for this view.",
+				ConflictsWith: []string{MAINTAINER_TEAM_KEY},
+			},
+			MAINTAINER_TEAM_KEY: {
+				Type:          schema.TypeString,
+				Optional:      true,
+				Description:   "The team key of the maintainer team for this view.",
+				ConflictsWith: []string{MAINTAINER_ID},
+			},
+			TAGS: tagsSchema(tagsSchemaOptions{isDataSource: false}),
+			ARCHIVED: {
+				Type:        schema.TypeBool,
+				Optional:    true,
+				Default:     false,
+				Description: "Whether the view is archived.",
+			},
+		},
+	}
+}
+
+func resourceViewCreate(ctx context.Context, d *schema.ResourceData, metaRaw interface{}) diag.Diagnostics {
+	client := metaRaw.(*Client)
+	betaClient, err := newBetaClient(client.apiKey, client.apiHost, false, DEFAULT_HTTP_TIMEOUT_S)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	projectKey := d.Get(PROJECT_KEY).(string)
+	viewKey := d.Get(KEY).(string)
+
+	viewPost := map[string]interface{}{
+		"key":  viewKey,
+		"name": d.Get(NAME).(string),
+	}
+
+	if description, ok := d.GetOk(DESCRIPTION); ok {
+		viewPost["description"] = description.(string)
+	}
+
+	if generateSdkKeys, ok := d.GetOk(GENERATE_SDK_KEYS); ok {
+		viewPost["generateSdkKeys"] = generateSdkKeys.(bool)
+	}
+
+	if maintainerId, ok := d.GetOk(MAINTAINER_ID); ok {
+		viewPost["maintainerId"] = maintainerId.(string)
+	}
+
+	if maintainerTeamKey, ok := d.GetOk(MAINTAINER_TEAM_KEY); ok {
+		viewPost["maintainerTeamKey"] = maintainerTeamKey.(string)
+	}
```

Comment:
you can but don't have to enforce at least one or the other of these but not both here - the API will fail if not

Reply:
nvm view_helper.go wasn't expanded

Reply:
okay actually sorry on second thought i think you should enforce these, because of the way we calculate it in gonfalon. if you define neither, the API will still return a maintainer_id, which i'm fairly certain will cause terraform to get confused about the state. there's no way to make maintainer_id optional but computed only when team_maintainer isn't defined, so i think the best way to do this is to hard enforce one or the other on the terraform side